test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -0,0 +1,1103 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+import curses
+import json
+import os
+
+
+class TestNonprintingCharacters(TestFunctional):
+    """
+    Test to check passing non-printable environment variables
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+
+        # Mapping of ASCII non-printable character to escaped representation
+        self.npcat = {
+            "\x00": "^@", "\x01": "^A", "\x02": "^B", "\x03": "^C",
+            "\x04": "^D", "\x05": "^E", "\x06": "^F", "\x07": "^G",
+            "\x08": "^H", "\x09": "^I", "\x0A": "^J", "\x0B": "^K",
+            "\x0C": "^L", "\x0D": "^M", "\x0E": "^N", "\x0F": "^O",
+            "\x10": "^P", "\x11": "^Q", "\x12": "^R", "\x13": "^S",
+            "\x14": "^T", "\x15": "^U", "\x16": "^V", "\x17": "^W",
+            "\x18": "^X", "\x19": "^Y", "\x1A": "^Z", "\x1B": "^[",
+            "\x1C": "^\\", "\x1D": "^]", "\x1E": "^^", "\x1F": "^_"
+        }
+        self.npch = self.npcat.keys()
```

Comment:
This variable is not necessary. You can just do...
```
if ch in self.npcat:
    print 'yes'
```

Reply:
Done.